### PR TITLE
Fix Clippy for the Rust 1.80 release

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/6114

# Rationale for this change
 
Rust 1.80 added a bunch of new lints which our codebase fails. Thus let's fix them

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-array/src/array/string_array.rs [hunk 1]
```diff
@@ -376,70 +376,6 @@ mod tests {
             .expect("All null array has valid array data");
     }
 
-    #[cfg(feature = "test_utils")]
```
Comment (alamb) [inline arrow-array/src/array/string_array.rs:379]: clippy flagged the fact that this crate doesn't have the `test_utils` feature (and thus this code is not reachable). So I deleted it

File: arrow-array/src/array/byte_view_array.rs [hunk 2]
```diff
@@ -66,7 +66,7 @@ use super::ByteArrayType;
 /// * Strings with length <= 12 are stored directly in the view.
 ///
 /// * Strings with length > 12: The first four bytes are stored inline in the
-/// view and the entire string is stored in one of the buffers.
+///   view and the entire string is stored in one of the buffers.
```
Comment (alamb) [inline arrow-array/src/array/byte_view_array.rs:69]: A new clippy lint requires items in docs to be properly indented

File: arrow/Cargo.toml [hunk 3]
```diff
@@ -75,7 +75,7 @@ pyarrow = ["pyo3", "ffi"]
 # force_validate runs full data validation for all arrays that are created
 # this is not enabled by default as it is too computationally expensive
 # but is run as part of our CI checks
-force_validate = ["arrow-data/force_validate"]
+force_validate = ["arrow-array/force_validate", "arrow-data/force_validate"]
```
Comment (alamb) [inline arrow/Cargo.toml:78]: There was code in `arrow-data` that was enabled by the `force_validate` config setting, but `arrow-data` doesn't actually have that feature

File: arrow-array/src/array/string_array.rs (hunk 1 above)
Comment (alamb) [reply arrow-array/src/array/string_array.rs:379]: This code also doesn't compile even when I added a `test_utils` feature flag -- I think it is left over from a time before we broke the crates apart

File: arrow-array/Cargo.toml [hunk 4]
```diff
@@ -52,6 +52,7 @@ hashbrown = { version = "0.14.2", default-features = false }
 
 [features]
 ffi = ["arrow-schema/ffi", "arrow-data/ffi"]
+force_validate = []
```
Comment (alamb) [inline arrow-array/Cargo.toml:55]: this crate has code that is flagged behind force_validate but it was not a crate feature

Comment (Xuanwo) [reply arrow-array/Cargo.toml:55]: checked cfg is really useful!

**Review by Dandandan — APPROVED** [review]

File: arrow-array/Cargo.toml (hunk 4 above)
Comment (alamb) [reply arrow-array/Cargo.toml:55]: 100% agree

**Pull request merged**